crawl-ref/source/annotate.cc

```diff
@@ -0,0 +1,240 @@
+/**
+ * @file
+ * @brief change the annotation of a dungeon level
+**/
+/* Known issues:
+ *   The includes need to be pruned down.
+ */
+#include "AppHdr.h"
+
+#include "annotate.h"
+
+#include <algorithm>
+#include <cctype>
+#include <cstdarg>
+#include <cstdio>
+#include <memory>
+#include <set>
+#include <sstream>
+
+#include "branch.h"
+#include "cloud.h"
+#include "clua.h"
+#include "command.h"
+#include "coordit.h"
+#include "daction-type.h"
+#include "dactions.h"
+#include "directn.h"
+#include "delay.h"
+#include "dgn-overview.h"
+#include "english.h"
+#include "env.h"
+#include "files.h"
+#include "food.h"
+#include "format.h"
+#include "god-abil.h"
+#include "god-passive.h"
+#include "hints.h"
+#include "item-name.h"
+#include "item-prop.h"
+#include "item-status-flag-type.h"
+#include "items.h"
+#include "libutil.h"
+#include "macro.h"
+#include "mapmark.h"
+#include "message.h"
+#include "mon-death.h"
+#include "nearby-danger.h"
+#include "output.h"
+#include "place.h"
+#include "prompt.h"
+#include "religion.h"
+#include "stairs.h"
+#include "state.h"
+#include "stringutil.h"
+#include "terrain.h"
+#include "tiles-build-specific.h"
+#include "traps.h"
+#include "travel.h"
+#include "unicode.h"
+#include "unwind.h"
+#include "view.h"
+
+// XXX The same 2 lines occur in gdn-overview.cc, so that this variable-type
+// is defined twice
+//typedef map<level_id, string> annotation_map_type;
+//annotation_map_type level_annotations;
```

Comment:
These should be uncommented, and you should add `extern` to the `level_annotations` line. Then you can remove the stuff in `dgn-overview.h`.

I'm guessing that you aded the stuff in `dgn-overview.h` because this code here failed with a linker error, e.g. "duplicate symbol: level_annotations". That's because the `annotations_map_type level_annotations` line in two different files will declare two different variables with the same name. If you add `extern` to one line, that line will then reference the variable that is now defined in only one place.